tests/integration/test_schema_registry_auth.py

```diff
@@ -29,139 +29,157 @@
 reader = aiohttp.BasicAuth("reader", "secret")
 
 
-async def test_sr_auth(registry_async_client_auth: Client) -> None:
+async def test_sr_auth(registry_async_retry_client_auth: RetryRestClient) -> None:
     subject = new_random_name("cave-")
 
-    res = await registry_async_client_auth.post(f"subjects/{quote(subject)}/versions", json={"schema": schema_avro_json})
+    res = await registry_async_retry_client_auth.post(
+        f"subjects/{quote(subject)}/versions", json={"schema": schema_avro_json}, expected_response_code=401
+    )
     assert res.status_code == 401
 
-    res = await registry_async_client_auth.post(
+    res = await registry_async_retry_client_auth.post(
         f"subjects/{quote(subject)}/versions", json={"schema": schema_avro_json}, auth=aladdin
     )
     assert res.status_code == 200
     sc_id = res.json()["id"]
     assert sc_id >= 0
 
-    res = await registry_async_client_auth.get(f"subjects/{quote(subject)}/versions/latest")
+    res = await registry_async_retry_client_auth.get(
+        f"subjects/{quote(subject)}/versions/latest", expected_response_code=401
+    )
     assert res.status_code == 401
```

Comment:
**[non-blocking]**
I know it'd be a harder refactor, but we also somehow already assert the response code:
```
if response.status_code != expected_response_code:
    raise UnexpectedResponseStatus(f"Unexpected status code: {response!r}")
```
we can omit the assertion `assert res.status_code == 401` in all places where we use the `retry_client` or even move the assertion into the client stub.

Reply:
Requires bit more work. I'd like to have this in first as these do reduce the flakiness.